sled-hardware/src/disk.rs

```diff
@@ -0,0 +1,397 @@
+        self.paths.partition_device_path(
+            &self.partitions,
+            Partition::BootImage,
+            raw,
+        )
+    }
+
+    pub fn slot(&self) -> i64 {
+        self.slot
+    }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+#[allow(dead_code)]
+pub enum DiskVariant {
+    U2,
+    M2,
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn test_disk_paths() {
+        const DEVFS_PATH: &'static str = "/devices/my/disk";
+        let paths =
+            DiskPaths { devfs_path: PathBuf::from(DEVFS_PATH), dev_path: None };
+        assert_eq!(
+            paths.whole_disk(false),
+            PathBuf::from(format!("{DEVFS_PATH}:wd"))
+        );
+        assert_eq!(
+            paths.whole_disk(true),
+            PathBuf::from(format!("{DEVFS_PATH}:wd,raw"))
+        );
+        assert_eq!(
+            paths.partition_path(0, false),
+            Some(PathBuf::from(format!("{DEVFS_PATH}:a")))
+        );
+        assert_eq!(
+            paths.partition_path(1, false),
+            Some(PathBuf::from(format!("{DEVFS_PATH}:b")))
+        );
+        assert_eq!(
+            paths.partition_path(2, false),
+            Some(PathBuf::from(format!("{DEVFS_PATH}:c")))
+        );
+        assert_eq!(
+            paths.partition_path(3, false),
+            Some(PathBuf::from(format!("{DEVFS_PATH}:d")))
+        );
+        assert_eq!(
+            paths.partition_path(4, false),
+            Some(PathBuf::from(format!("{DEVFS_PATH}:e")))
+        );
+        assert_eq!(
+            paths.partition_path(5, false),
+            Some(PathBuf::from(format!("{DEVFS_PATH}:f")))
+        );
+        assert_eq!(paths.partition_path(6, false), None);
+
+        assert_eq!(
+            paths.partition_path(0, true),
+            Some(PathBuf::from(format!("{DEVFS_PATH}:a,raw")))
+        );
+        assert_eq!(
+            paths.partition_path(1, true),
+            Some(PathBuf::from(format!("{DEVFS_PATH}:b,raw")))
+        );
+        assert_eq!(
+            paths.partition_path(2, true),
+            Some(PathBuf::from(format!("{DEVFS_PATH}:c,raw")))
+        );
+        assert_eq!(
+            paths.partition_path(3, true),
+            Some(PathBuf::from(format!("{DEVFS_PATH}:d,raw")))
+        );
+        assert_eq!(
+            paths.partition_path(4, true),
+            Some(PathBuf::from(format!("{DEVFS_PATH}:e,raw")))
+        );
+        assert_eq!(
+            paths.partition_path(5, true),
+            Some(PathBuf::from(format!("{DEVFS_PATH}:f,raw")))
+        );
+        assert_eq!(paths.partition_path(6, true), None);
+    }
+
+    #[test]
+    fn test_partition_device_paths() {
+        const DEVFS_PATH: &'static str = "/devices/my/disk";
+        let paths =
+            DiskPaths { devfs_path: PathBuf::from(DEVFS_PATH), dev_path: None };
+
+        assert_eq!(
+            paths
+                .partition_device_path(
+                    &[Partition::ZfsPool,],
```

Comment:
Trivial nit that existed before this PR:

```suggestion
                    &[Partition::ZfsPool],
```

Reply:
Done